src/Event/EventManager.php

```diff
@@ -442,6 +442,24 @@ public function prioritisedListeners($eventKey)
     }
 
     /**
+     * Returns the listeners matching a specified pattern
+     *
+     * @param string $eventKeyPattern Pattern to match.
+     * @return array
+     */
+    public function matchingListeners($eventKeyPattern)
+    {
+        $matchPattern = '/' . preg_quote($eventKeyPattern, "/") . '/';
```

Comment:
@markstory Apologies for bringing this topic up again. 

I've made the change to ``EventManager.php`` (locally) to add anchors to the ``$matchPattern`` after running the ``$eventKeyPattern`` through the ``preg_quote()`` function (see below) and started changing the accompanying unit-tests in ``EventManagerTest.php``, but I have to say this feels so limiting versus the implemenation I presently have.

The implementation **without** anchors allows the developer to search through the existing listeners array for a full or partial match on the pattern argument. I refer you to the unit-tests associated with this enhancement that I have added in [EventManagerTest.php](https://github.com/cakephp/cakephp/pull/8104/files#diff-b0d390b37d28dbd12326d0b715d909d7).

Since the ``preg_quote()`` is scrubbing the pattern argument of any regex characters I think the complexity associated with regex implementation greatly simplified.

**Questions, comments, or observations?**

    public function matchingListeners($eventKeyPattern)
    {
        $matchPattern = '/^' . preg_quote($eventKeyPattern, "/") . '$/';
        . . .